src/de/mod.rs

```diff
@@ -82,10 +83,34 @@ impl<'de> Deserializer<'de> {
 impl<'de, 'a> de::Deserializer<'de> for &'a mut Deserializer<'de> {
     type Error = Error;
 
-    fn deserialize_any<V>(self, _visitor: V) -> Result<V::Value>
+    fn deserialize_any<V>(self, visitor: V) -> Result<V::Value>
         where V: Visitor<'de>
     {
-        panic!("Give me some!");
+        if self.bytes.consume_ident("true") {
+            return visitor.visit_bool(true);
+        } else if self.bytes.consume_ident("false") {
+            return visitor.visit_bool(false);
+        } else if self.bytes.check_ident("Some") {
+            return self.deserialize_option(visitor);
+        } else if self.bytes.consume_ident("None") {
+            return visitor.visit_none();
+        } else if self.bytes.consume("()") {
+            return visitor.visit_unit();
+        }
+
+        if self.bytes.identifier().is_ok() {
```

Comment:
This is a little bit difficult.. I want to ignore this if it's e.g. the name of a struct, but if we're deserializing a struct, that's a problem (because the field names aren't strings like in JSON, but just unquoted identifiers). This would mean I need context information :(

Reply:
@dtolnay Do you have an advice how I could solve that?

Reply:
Here are 3 ways to store context information:

- In `self`.
- Passed as function arguments, for example save the identifier and pass it to the function responsible for deserializing a struct.
- In a new Deserializer that wraps `self` and has additional fields.